app/src/main/java/co/condorlabs/customcomponents/helper/masks/PhoneNumberTextWatcherMask.kt

```diff
@@ -19,51 +19,55 @@ package co.condorlabs.customcomponents.helper.masks
+            val hadParenthesis = text.toString().firstOrNull() == CHAR_OPENING_PARENTHESIS
+            var mask = 0
+            var result = text.toString()
+                .replace(PHONE_NUMBER_SEPARATOR_TOKEN, "")
+                .replace(OPENING_PARENTHESIS, "")
+                .replace(CLOSING_PARENTHESIS, "")
 
-            when (resultado.length) {
+            when (result.length) {
                 in PHONE_NUMBER_REGEX_FIRST_GROUP_RANGE_BOTTOM..PHONE_NUMBER_REGEX_FIRST_GROUP_RANGE_TOP -> {
-                    resultado = resultado.replaceFirst(
+                    result = result.replaceFirst(
                         PHONE_NUMBER_REGEX_FIRST_AND_SECOND_GROUP_MATCHER.toRegex(),
                         PHONE_NUMBER_REGEX_FIRST_GROUP_REPLACEMENT_MATCHER
                     )
+                    mask = 1
                 }
                 in PHONE_NUMBER_REGEX_SECOND_GROUP_RANGE_BOTTOM..PHONE_NUMBER_REGEX_SECOND_GROUP_RANGE_TOP -> {
-                    resultado = resultado.replaceFirst(
-                        "$PHONE_NUMBER_REGEX_FIRST_AND_SECOND_GROUP_MATCHER$PHONE_NUMBER_REGEX_FIRST_AND_SECOND_GROUP_MATCHER".toRegex(),
+                    result = result.replaceFirst(
+                        "$PHONE_NUMBER_REGEX_FIRST_AND_SECOND_GROUP_MATCHER$PHONE_NUMBER_REGEX_SECOND_GROUP_MATCHER".toRegex(),
                         PHONE_NUMBER_REGEX_SECOND_GROUP_REPLACEMENT_MATCHER
                     )
+                    mask = 2
                 }
                 in PHONE_NUMBER_REGEX_THIRD_GROUP_RANGE_BOTTOM..PHONE_NUMBER_REGEX_THIRD_GROUP_RANGE_TOP -> {
-                    resultado = resultado.replaceFirst(
-                        ("$PHONE_NUMBER_REGEX_FIRST_AND_SECOND_GROUP_MATCHER" +
-                                "$PHONE_NUMBER_REGEX_FIRST_AND_SECOND_GROUP_MATCHER" +
-                                "$PHONE_NUMBER_REGEX_THIRD_GROUP_MATCHER").toRegex(),
+                    result = result.replaceFirst(
+                        ("$PHONE_NUMBER_REGEX_FIRST_AND_SECOND_GROUP_MATCHER$PHONE_NUMBER_REGEX_SECOND_GROUP_MATCHER$PHONE_NUMBER_REGEX_THIRD_GROUP_MATCHER").toRegex(),
                         PHONE_NUMBER_REGEX_THIRD_GROUP_REPLACEMENT_MATCHER
                     )
+                    mask = 3
                 }
             }
 
-            text.replace(FIRST_EDITTEXT_SELECTION_CHARACTER, text.length, resultado)
+            text.replace(FIRST_EDITTEXT_SELECTION_CHARACTER, text.length, result)
 
-            mReceiver.addTextChangedListener(this)
-        }
-    }
-
-    override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
-
-        s?.let {
-            if ((it.length == PHONE_NUMBER_FORMAT_FIRST_HYPHEN_INDEX ||
-                        it.length == PHONE_NUMBER_FORMAT_SECOND_HYPHEN_INDEX)
-            ) {
-                mReceiver.append(HYPHEN)
+            receiver.addTextChangedListener(this)
+            if (hadParenthesis and (mask == 1)) {
```

Comment:
Why `mask` is number type? If it is only used when is equals to 1, it should be a boolean.

Reply:
done